src/openvic-simulation/economy/trading/MarketInstance.cpp

```diff
@@ -0,0 +1,37 @@
+#include "MarketInstance.hpp"
+
+#include "openvic-simulation/economy/GoodDefinition.hpp"
+#include "openvic-simulation/economy/GoodInstance.hpp"
+#include "openvic-simulation/economy/trading/MarketSellOrder.hpp"
+
+using namespace OpenVic;
+
+bool MarketInstance::setup(GoodInstanceManager& new_good_instance_manager) {
+	good_instance_manager = &new_good_instance_manager;
+	return true;
+}
+
+void MarketInstance::place_market_sell_order(const MarketSellOrder market_sell_order) {
+	GoodDefinition const* const good = market_sell_order.get_good();
+	GoodInstance* const good_instance = good_instance_manager->get_good_instance_by_identifier(good->get_identifier());
+	good_instance->add_market_sell_order(market_sell_order.get_quantity(), market_sell_order.get_after_trade());
+}
+
+void MarketInstance::execute_orders() {
+	std::vector<GoodInstance>& good_instances = good_instance_manager->get_good_instances();
+	for (GoodInstance& good_instance : good_instances) {
```

Comment:
Move the contents of this loop to GoodInstance?

Reply:
I moved the contents. That way the vectors can remain private for the goodinstance.